lib/galaxy/datatypes/binary.py

```diff
@@ -1047,6 +1047,47 @@ def sniff(self, filename):
             return False
 
 
+class HIC(Binary):
+    """
+    Class describing an Juicer hic file
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('SRR1791297_30.hic')
+    >>> HIC().sniff(fname)
+    True
+    """
+    file_ext = "hic"
+    # edam_format = "format_3590" # Don't know what to set here or if it is necessary at all. Copied from h5(binary) class.
+
+    def __init__(self, **kwd):
+        Binary.__init__(self, **kwd)
+        self._magic = b'HIC'
+
+    def sniff(self, filename):
+        try:
+            header = open(filename, 'rb')
+            header_magic = struct.unpack('<3s', header.read(3))[0]
+            if header_magic == self._magic:
+                return True
+            return False
+        except Exception:
+            return False
+
+    def set_peek(self, dataset, is_multi_byte=False):
```

Comment:
```suggestion
    def set_peek(self, dataset: "DatasetInstance", is_multi_byte: bool = False, **kwd) -> None:
```
Even though  you don't need the `**kwd` for this method, you need to include it, otherwise type-checking will raise an error (this class is a subclass of Binary, which defines this method with `**kwd`)